### PR TITLE
Fix torwards ranked score being awarded for all scores

### PR DESCRIPTION
**Pull request by lenforiee**

### CODE REVIEW

File: handlers.py [hunk 1]
```diff
@@ -554,20 +566,25 @@ async def handle_submission(request: Request) -> Response:
             stats.total_score += score.score
             stats.total_hits += score.n300 + score.n100 + score.n50
 
-            additive = score.score
-            if score.previous_score and score.status == 3:
-                additive -= score.previous_score.score
+            if score.mode.as_mode_int() in (1, 3):
```
Comment (cmyui) [inline handlers.py:557]: why only taiko & mania?

Comment (lenforiee) [reply handlers.py:557]: that was my addition that i found in bancho.py basically geki and katu are added to total hits for mania and taiko (you had the comment explaining it there)

Comment (cmyui) [reply handlers.py:557]: ah right, didn't notice this was only adding geki & katu, nice

Comment (cmyui) [inline handlers.py:557]:
```suggestion
            if score.mode.as_mode_int() in (1, 3):
                # taiko uses geki & katu for hitting big notes with 2 keys
                # mania uses geki & katu for rainbow 300 & 200
```

File: handlers.py [hunk 2]
```diff
@@ -554,20 +566,27 @@ async def handle_submission(request: Request) -> Response:
...
+                # taiko uses geki & katu for hitting big notes with 2 keys
+                # mania uses geki & katu for rainbow 300 & 200
+                stats.total_hits += score.geki + score.katu
 
             if score.passed and score.map.has_leaderboard:
-                if score.map.status == 2:
-                    stats.ranked_score += additive
+                
+                if score.status == 3 and score.map.status == 2:
```
Comment (cmyui) [inline handlers.py]: we should replace these int enum values with a named enum - this is just annoying to read lol

**Review by cmyui — APPROVED** [review]

**Pull request merged**